### PR TITLE
Add lint check for min GPA

### PR DESCRIPTION
**Pull request by josueetcom**
## Motivation and Context

Addresses one of tasks in #858.

## How Has This Been Tested?

Added tests for the new files. The card tests are missing but the card view has been tested locally.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] User visible change (users will notice UI or functional changes)

## Previewing Changes

1. Go to Render preview URL.
2. Try adding/editing a scholarship so that the minimum GPA is mentioned in the description.
3. See the error on the card.
4. Log out and make sure you can't see the error.

## Checklist:

- [x] I have added tests to cover my changes.

### CODE REVIEW

File: src/components/ScholarshipCard.jsx [hunk 1]
```diff
@@ -107,6 +110,7 @@ export default function ScholarshipCard({ scholarship, style }) {
   }, [author, currentUser, preview]);
 
   const CardAreaComponent = detailed ? Box : CardActionArea;
+  const lintIssues = canEdit ? lint(scholarship.data) : [];
```
Comment (edwingl10) [inline src/components/ScholarshipCard.jsx:113]: It would also be nice to see the warnings when adding a scholarship.

File: src/lib/lint.test.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      'GPAs of 3.0-3.5 encouraged to apply.',
+    ].forEach((d) => expect(parseMinGPA(d)?.value).toBe('3.0')));
+
+  test('matches minimum GPA value', () =>
+    [
+      'Min GPA of 3.0 out of 4.0',
+      'Min GPA: High school 3.5, College 3.0',
+      'Must have a GPA of 3.0 out of 4.0 or 4.0 out of 5.0',
+      'GPAs of 3.0-3.5 encouraged to apply.',
+    ].forEach((d) => expect(parseMinGPA(d)?.value).toBe('3.0')));
```
Comment (edwingl10) [inline src/lib/lint.test.ts:39]: same test as above

Comment (edwingl10) [inline src/lib/lint.test.ts:39]: same test as above

**Review by edwingl10 — APPROVED** [review]

File: src/components/ScholarshipCard.jsx (hunk 1 above)
Comment (josueetcom) [reply src/components/ScholarshipCard.jsx]: oh, I hadn't noticed it didn't appear for new scholarships. Fixed.

**Pull request merged**